### ISSUE TITLE
[Feature request] run du -sh to prevent DB corruption

### ISSUE THREAD

**Issue by djvs** [neo4j/neo4j#5737]
One event of the hard disk running out of space on my server was enough to mess up my server pretty much permanently (still waiting to get Michael Hunger's store-utils tool to work at all).  I was surprised that the DB would break itself when the disk runs out of space.  Maybe you could at least run du -sh once an hour and throw some errors into the logs if someone tried to write to the db when the disk didn't have space above a certain threshold?

**Comment by cleishm** [comment]
Hi @djvs,

We'll certainly look into better ways to alert if disk space appears to be critical. Running `du -sh` may be a way to do that, although I'm aware that there are a lot of problems with trying to determine available disk space reliably across platforms, especially when using virtual servers and elastic storage solutions.

However, we would like to better understand the "permanent" error that you've seen result from running out of disk space. If you can supply us with a process to reproduce, I'm confident our QA team would love to look into that further.

Cheers,
Chris

**Comment by djvs** [comment]
Hi @cleishm,

Sure, the suggestion is just a general one.  

The database corruption issue is here:

https://github.com/neo4j/neo4j/issues/5735

although, while I suspect it was caused by that one instance (the only time I had serious trouble getting the server to restart), I can't be certain.  You can see that some strange behavior is exhibited - rels pulled with overlapping IDs, a "null" result in neo4j-shell...

**Comment by djvs** [comment]
I also have consistency check failure information if needed...plus I can tell you that `neo4j-shell -v -c dump > export_data.cypher` returns this:

```
ERROR (-v for expanded information):
    null
java.lang.IndexOutOfBoundsException
    at org.neo4j.io.pagecache.impl.muninn.MuninnPageCursor.setOffset(MuninnPageCursor.java:419)
    at org.neo4j.kernel.impl.store.RelationshipGroupStore.readRecord(RelationshipGroupStore.java:143)
    at org.neo4j.kernel.impl.store.RelationshipGroupStore.forceGetRecord(RelationshipGroupStore.java:106)
    at org.neo4j.kernel.impl.api.store.StoreNodeRelationshipCursor.init(StoreNodeRelationshipCursor.java:89)
    at org.neo4j.kernel.impl.api.store.StoreAbstractNodeCursor.relationships(StoreAbstractNodeCursor.java:196)
    at org.neo4j.kernel.api.cursor.NodeItem$NodeItemHelper.getRelationships(NodeItem.java:87)
    at org.neo4j.kernel.impl.api.OperationsFacade.nodeGetRelationships(OperationsFacade.java:330)
    at org.neo4j.kernel.impl.core.NodeProxy$1.iterator(NodeProxy.java:154)
    at org.neo4j.kernel.impl.core.NodeProxy$1.iterator(NodeProxy.java:146)
    at org.neo4j.cypher.export.SubGraphExporter.appendRelationships(SubGraphExporter.java:162)
    at org.neo4j.cypher.export.SubGraphExporter.export(SubGraphExporter.java:56)
    at org.neo4j.shell.kernel.apps.cypher.Exporter.export(Exporter.java:43)
    at org.neo4j.shell.kernel.apps.cypher.Dump.export(Dump.java:88)
    at org.neo4j.shell.kernel.apps.cypher.Dump.exec(Dump.java:58)
    at org.neo4j.shell.kernel.apps.TransactionProvidingApp.execute(TransactionProvidingApp.java:253)
    at org.neo4j.shell.impl.AbstractAppServer.interpretLine(AbstractAppServer.java:126)
    at org.neo4j.shell.kernel.GraphDatabaseShellServer.interpretLine(GraphDatabaseShellServer.java:96)
    at org.neo4j.shell.impl.RemotelyAvailableServer.interpretLine(RemotelyAvailableServer.java:60)
    at sun.reflect.GeneratedMethodAccessor106.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:322)
    at sun.rmi.transport.Transport$2.run(Transport.java:202)
    at sun.rmi.transport.Transport$2.run(Transport.java:199)
    at java.security.AccessController.doPrivileged(Native Method)
    at sun.rmi.transport.Transport.serviceCall(Transport.java:198)
    at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:567)
    at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
    at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.access$400(TCPTransport.java:619)
    at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:684)
    at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$1.run(TCPTransport.java:681)
    at java.security.AccessController.doPrivileged(Native Method)
    at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:681)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
```

**Comment by chrisvest** [comment]
The bounds error in `RelationshipGroupStore` has already been fixed.

**Comment by djvs** [comment]
chrisvest: Is there a release for that or is it something I have to compile?

**Comment by chrisvest** [comment]
The fix #5694 will be in 2.3.1, which as far as I'm aware doesn't have a date yet.
